### PR TITLE
Number picker component

### PR DESCRIPTION
**Pull request by Saionaro**

### CODE REVIEW

File: components/NumberPicker/NumberPicker.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  component: Component,
+  title: "NumberPicker",
+};
+
+export const Slider = StoryComponent.bind({});
+
+Slider.args = {
+  label: "Likes",
+  initialValue: 15,
+  onChange: () => console.log,
```
Comment (iAdramelk) [inline components/NumberPicker/NumberPicker.stories.tsx:28]: Please use standard actions instead https://storybook.js.org/docs/react/essentials/actions#action-args

File: components/NumberPicker/NumberPicker.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  renderMax: (val) => `${val}++`,
+};
```
Comment (iAdramelk) [inline components/NumberPicker/NumberPicker.stories.tsx:30]: Can you add examples with min values and other fields so we can change them in docs tab?

File: components/NumberPicker/Slider.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  height: 22,
+  boxShadow: "0 0 4px #D2DBDF",
+  transition: transition([["backgroundColor", "interaction"]]),
+  cursor: "ew-resize",
+});
+
+const StyledRange = styled("input")<StyledSystemProps>(
+  css({
+    width: "100%",
+    height: 10,
```
Comment (iAdramelk) [inline components/NumberPicker/Slider.tsx:102]:
```suggestion
    height: "10px",
```

I think we will add 10th element to the theme sooner or later.

File: components/NumberPicker/Slider.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    ":active, :focus": {
+      "&::-webkit-slider-thumb": {
+        backgroundColor: "lightest-gray",
+      },
+      "&::-moz-range-thumb": {
+        backgroundColor: "lightest-gray",
+      },
+    },
+  }),
+  all
```
Comment (iAdramelk) [inline components/NumberPicker/Slider.tsx:131]: Do we use this `all` anywhere?

File: components/NumberPicker/Slider.tsx [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    color: "gray",
+    fontSize: "text-sm",
+    lineHeight: "12px",
+  }),
+  all
+);
+
+const thumbStyles = css({
+  appearance: "none",
+  backgroundColor: "white",
```
Comment (iAdramelk) [inline components/NumberPicker/Slider.tsx:88]:
```suggestion
  bg: "white",
```

File: components/NumberPicker/Slider.tsx [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+  cursor: "ew-resize",
+});
+
+const StyledRange = styled("input")<StyledSystemProps>(
+  css({
+    width: "100%",
+    height: 10,
+    my: "6px",
+    appearance: "none",
+    backgroundColor: "white",
```
Comment (iAdramelk) [inline components/NumberPicker/Slider.tsx:105]:
```suggestion
    bg: "white",
```

File: components/NumberPicker/Slider.tsx [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+    border: "1px solid",
+    borderColor: "light-gray",
+    borderRadius: "circle",
+    cursor: "pointer",
+    boxShadow: "0 0 4px #D2DBDF",
+    "&::-webkit-slider-thumb": thumbStyles,
+    "&::-moz-range-thumb": thumbStyles,
+    "&::-webkit-progress-value": {
+      appearance: "none",
+      backgroundColor: "dark-purple",
```
Comment (iAdramelk) [inline components/NumberPicker/Slider.tsx:115]:
```suggestion
      bg: "dark-purple",
```

File: components/NumberPicker/Slider.tsx [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+    },
+    "&::-moz-range-progress": {
+      backgroundColor: "dark-purple",
```
Comment (iAdramelk) [inline components/NumberPicker/Slider.tsx:118]:
```suggestion
      bg: "dark-purple",
```

File: components/NumberPicker/Slider.tsx [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+      borderRadius: "circle",
+      height: "100%",
+    },
+    ":active, :focus": {
+      "&::-webkit-slider-thumb": {
+        backgroundColor: "lightest-gray",
```
Comment (iAdramelk) [inline components/NumberPicker/Slider.tsx:124]:
```suggestion
        bg: "lightest-gray",
```

File: components/NumberPicker/Slider.tsx [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+      },
+      "&::-moz-range-thumb": {
+        backgroundColor: "lightest-gray",
```
Comment (iAdramelk) [inline components/NumberPicker/Slider.tsx:127]:
```suggestion
        bg: "lightest-gray",
```

**Review by iAdramelk — CHANGES_REQUESTED** [review]

File: components/NumberPicker/Slider.tsx (hunk 3 above)
Comment (Saionaro) [reply components/NumberPicker/Slider.tsx:102]: After some experiments I realized that `height` and `width` doesn't resolve `space` from styled system. ¯\_(ツ)_/¯
It just translates number to pixels value

File: components/NumberPicker/NumberPicker.stories.tsx (hunk 2 above)
Comment (Saionaro) [reply components/NumberPicker/NumberPicker.stories.tsx:30]: We can change even not specified props
Screenshot 2021-04-20 at 15 28 32

File: components/NumberPicker/Slider.tsx (hunk 4 above)
Comment (Saionaro) [reply components/NumberPicker/Slider.tsx:131]: Actually great catch 😅

**Pull request merged**